controllers/edgedeployment_controller.go

```diff
@@ -367,3 +381,60 @@ func splitEdgeDevices(edgeDevices []managementv1alpha1.EdgeDevice, splitSize uin
 	}
 	return result
 }
+
+func (r *EdgeDeploymentReconciler) updateLabelsFromSelector(ctx context.Context, edgeDeployment *managementv1alpha1.EdgeDeployment) error {
+	edgeDeploymentCopy := edgeDeployment.DeepCopy()
+	UpdateSelectorLabels(edgeDeploymentCopy)
+	newLabels := edgeDeploymentCopy.Labels
+
+	if (len(newLabels) == 0 && edgeDeployment.Labels == nil) || reflect.DeepEqual(newLabels, edgeDeployment.Labels) {
+		return nil
+	}
+
+	err := r.EdgeDeploymentRepository.Patch(ctx, edgeDeployment, edgeDeploymentCopy)
+	return err
+}
+
+func isSelectorLabel(label string) bool {
+	return strings.HasPrefix(label, selectorLabelPrefix)
+}
+
+func CreateSelectorLabel(label string) string {
```

Comment:
This should be private.

Reply:
I use it in tests. Tests are a separate package.

Reply:
In that case it looks like a good candidate for moving to https://github.com/jakub-dzon/k4e-operator/blob/8081cb355b8c1daa5af3e15cb85ea92ea00e2093/internal/labels/labels.go

Reply:
Sure, I'll move it. But, I think we have too many files and packages. I worry that this way people won't find the code they can reuse. I personally, prefer larger files and packages that include a lot. Unless there's some logic for the separation.